src/main/java/gregtech/api/pipenet/block/BlockPipe.java

```diff
@@ -371,79 +380,36 @@ protected boolean isThisPipeBlock(Block block) {
         return null;
     }
 
-    /**
-     * Tests whatever pipe at given position can connect to a pipe from fromFacing face with fromColor color
-     * Used purely for rendering purposes
-     *
-     * @return 0 - not a pipe; 1 - pipe but blocked; 2,3 - accessible
-     */
-    protected final int isPipeAccessibleAtSide(IBlockAccess world, IPipeTile<PipeType, NodeDataType> selfTile, EnumFacing side) {
-        IPipeTile<PipeType, NodeDataType> tileEntityPipe = getPipeTileEntity(world, selfTile.getPipePos().offset(side));
-        return isPipeAccessibleAtSideInternal(selfTile, tileEntityPipe, side);
+    public boolean canConnect(IPipeTile<PipeType, NodeDataType> selfTile, EnumFacing facing) {
+        if(selfTile.getPipeWorld().getBlockState(selfTile.getPipePos().offset(facing)).getBlock() == Blocks.AIR) return false;
+        TileEntity other = selfTile.getPipeWorld().getTileEntity(selfTile.getPipePos().offset(facing));
+        if (other instanceof IPipeTile)
+            return canPipesConnect(selfTile, facing, (IPipeTile<PipeType, NodeDataType>) other);
+        return canPipeConnectToBlock(selfTile, facing, other);
```

Comment:
```suggestion
        return canPipeConnectToBlock(selfTile, facing.getOpposite(), other);
```
Fixes an issue where sided capabilities will not properly connect. For example, the singleblock generators, or the dynamo hatch